flytekit/core/workflow.py

```diff
@@ -731,7 +731,10 @@ def wrapper(fn):
             default_metadata=workflow_metadata_defaults,
             docstring=Docstring(callable_=fn),
         )
-        workflow_instance.compile()
+
+        ctx = FlyteContextManager.current_context()
+        if ctx.execution_state.mode != ctx.execution_state.Mode.TASK_EXECUTION:
```

Comment:
cc @kumare3 could we just skip it in `TASK_EXECUTION` mode (running the workflow on flyte cluster)?